arbor/include/arbor/event_generator.hpp

```diff
@@ -68,9 +68,10 @@ struct empty_generator {
     event_seq events(time_type, time_type) {
         return {nullptr, nullptr};
     }
-    std::vector<cell_lid_type> targets() {
+    std::vector<cell_local_label_type> targets() {
         return {};
     };
+    void init(const std::vector<cell_lid_type>&) {};
```

Comment:
Regarding the different semantics between the schedule-based and explicit generators, I have changed it so that both will cycle through the range of possible lids according to the selection policy.